modyn/models/articlenet/articlenet.py

```diff
@@ -0,0 +1,54 @@
+from typing import Any
+
+import torch
+from torch import nn
+from transformers import DistilBertModel
+
+
+class ArticleNet:
+    """
+    Adapted from WildTime. This network is used for NLP tasks (Arxiv and Huffpost)
+    Here you can find the original implementation:
+    https://github.com/huaxiuyao/Wild-Time/blob/main/wildtime/networks/article.py
+    """
+
+    # pylint: disable-next=unused-argument
+    def __init__(self, model_configuration: dict[str, Any], device: str, amp: bool) -> None:
+        self.model = ArticleNetwork(**model_configuration)
+        self.model.to(device)
+
+
+class DistilBertFeaturizer(DistilBertModel):
+    def __init__(self, config: Any) -> None:
+        super().__init__(config)
+        self.d_out = config.hidden_size
+
+    def __call__(self, data: torch.Tensor) -> torch.Tensor:
+        # slice the input tensor to get input ids and attention mask
+        # The model receives as input the output of the tokenizer, where the first dimension
+        # contains the tokens and the second a boolean mask to indicate which tokens are valid
+        # (the sentences have different lengths but the output of the tokenizer has always the same size,
+        # so you need the mask to understand what is useful data and what is just padding)
+        input_ids = data[:, :, 0]
+        attention_mask = data[:, :, 1]
+        # DistilBert's forward pass
+        hidden_state = super().__call__(
+            input_ids=input_ids,
+            attention_mask=attention_mask,
+        )[
+            0
+        ]  # 0: last hidden state, 1: hiddent states, 2: attentions
+        pooled_output = hidden_state[:, 0]  # first token is the pooled output, which is the aggregated representation
```

Comment:
just for my understanding: distilbert's output is redundant in the sense that we have pooled on idx 0 and then the individual representations again?